src/ol/interaction/Select.js

```diff
@@ -345,28 +343,44 @@ class Select extends Interaction {
   removeFeature_(evt) {
     const feature = evt.element;
     if (this.style_) {
-      this.removeSelectedStyle_(feature);
+      this.restorePreviousStyle_(feature);
     }
   }
 
+  /**
+   * @return {import("../style/Style.js").default|Array.<import("../style/Style.js").default>|import("../style/Style.js").StyleFunction|null} Select style.
+   */
+  getStyle() {
+    return this.style_;
+  }
+
   /**
    * @param {import("../Feature.js").default} feature Feature
    * @private
    */
-  giveSelectedStyle_(feature) {
+  applySelectedStyle_(feature) {
     const key = getUid(feature);
-    this.featureStyleAssociation_[key] = feature.getStyle();
+    if (!(key in originalFeatureStyles)) {
+      originalFeatureStyles[key] = feature.getStyle();
+    }
     feature.setStyle(this.style_);
   }
 
   /**
    * @param {import("../Feature.js").default} feature Feature
    * @private
    */
-  removeSelectedStyle_(feature) {
+  restorePreviousStyle_(feature) {
     const key = getUid(feature);
-    feature.setStyle(this.featureStyleAssociation_[key]);
-    delete this.featureStyleAssociation_[key];
+    const selectInteractions = /** @type {Array<Select>} */ (this.getMap().getInteractions().getArray().filter(function(interaction) {
+      return interaction instanceof Select && interaction.getFeatures().getArray().indexOf(feature) !== -1;
```

Comment:
Thanks @ahocevar for this PR. It seems to me that the filter condition should check if the feature has style.
```javascript
return interaction instanceof Select && interaction.getStyle() && interaction.getFeatures().getArray().indexOf(feature) !== -1;
```

Reply:
@michalzielanski `interaction.getStyle() !== false` to be precise:

> `style`: Style for the selected features. By default the default edit style is used (see module:ol/style). If set to false the selected feature's style will not change.

Can you create a follow-up pull request?

Reply:
In other places there is no strong type checking. There is only `if (this.style_)`, so `null` instead of `false` will also work. Using `interaction.getStyle() !== false` only here will cause a malfunction if someone uses `null` instead of `false`.
Creating PR would take me half a day, you will do it quicker :)

Reply:
See #10708. Feel free to buy me a beer @michalzielanski ;-)

Reply:
:)